Unsupported value: "tcp"

I'm not 100% sure it's a bug here, but since updating minikube and kubectl, tcp is not longer supported.
Must be TCP/UDP, resulting in the below error when creating the service.

### Description
[INFO]
[INFO] <<< fabric8-maven-plugin:3.1.79:deploy (default-cli) < install @ spring-boot-hystrix <<<
[INFO]
[INFO] --- fabric8-maven-plugin:3.1.79:deploy (default-cli) @ spring-boot-hystrix ---
[INFO] F8: Using Kubernetes at https://192.168.99.100:8443/ in namespace default with manifest /Users/taariq/code/fabric8-qs/spring-boot-hystrix/target/classes/META-INF/fabric8/kubernetes.yml
[INFO] Using namespace: default
[INFO] Creating a Service from kubernetes.yml namespace default name spring-boot-hystrix
[ERROR] Failed to create Service from kubernetes.yml. io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://192.168.99.100:8443/api/v1/namespaces/default/services. Message: Service "spring-boot-hystrix" is invalid: [spec.ports[0].protocol: Unsupported value: "tcp": supported values: TCP, UDP, spec.ports[0].protocol: Invalid value: "tcp": cannot create an external load balancer with non-TCP/UDP ports].

### Info

* fabric8-maven-plugin version: 3.1.79
* Maven version (`mvn -v`) : 3.3.9
* minikube version: v0.12.0
* kubernetes version : v1.4.3 (client and server)
* If it's a bug, how to reproduce: 
Clone the following sample project and deploy to kubernetes
* Sample project: *https://github.com/fabric8-quickstarts/spring-boot-hystrix*

To workaround I added a service.yaml to make it TCP, eg;

```
spec:
  ports:
  - name: "http"
    port: 80
    protocol: "TCP"

```

--- a/core/src/main/java/io/fabric8/maven/core/config/ServiceConfig.java
+++ b/core/src/main/java/io/fabric8/maven/core/config/ServiceConfig.java
@@ -109,7 +109,7 @@ public class ServiceConfig {
 
         // Protocol to use. Can be either "tcp" or "udp"
         @Parameter
-        ServiceProtocol protocol;
+        String protocol;
 
         // Container port to expose
         @Parameter
@@ -128,7 +128,7 @@ public class ServiceConfig {
         String name;
 
         public ServiceProtocol getProtocol() {
-            return protocol;
+            return protocol != null ? ServiceProtocol.valueOf(protocol.toUpperCase()) : null;
         }
 
         public int getPort() {
@@ -158,6 +158,11 @@ public class ServiceConfig {
             }
 
             public Builder protocol(ServiceProtocol protocol) {
+                config.protocol = protocol != null ? protocol.name() : null;
+                return this;
+            }
+
+            public Builder protocol(String protocol) {
                 config.protocol = protocol;
                 return this;
             }

--- a/core/src/main/java/io/fabric8/maven/core/config/ServiceProtocol.java
+++ b/core/src/main/java/io/fabric8/maven/core/config/ServiceProtocol.java
@@ -20,5 +20,5 @@ package io.fabric8.maven.core.config;
  * @since 22/03/16
  */
 public enum ServiceProtocol {
-    udp, tcp;
+    UDP, TCP;
 }

--- a/core/src/main/java/io/fabric8/maven/core/handler/ServiceHandler.java
+++ b/core/src/main/java/io/fabric8/maven/core/handler/ServiceHandler.java
@@ -69,7 +69,7 @@ public class ServiceHandler {
             for (ServiceConfig.Port port : service.getPorts()) {
                 ServicePort servicePort = new ServicePortBuilder()
                     .withName(port.getName())
-                    .withProtocol(port.getProtocol() != null ? port.getProtocol().name() : "tcp")
+                    .withProtocol(port.getProtocol() != null ? port.getProtocol().name() : "TCP")
                     .withTargetPort(new IntOrString(port.getTargetPort()))
                     .withPort(port.getPort())
                     .withNodePort(port.getNodePort())

--- a/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java
+++ b/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java
@@ -165,11 +165,11 @@ public class DefaultServiceEnricher extends BaseEnricher {
                         }
                         ret.add(
                                 new ServiceConfig.Port.Builder()
-                                        .protocol(ServiceProtocol.tcp) // TODO: default for the moment
-                                        .port(servicePort)
-                                        .targetPort(portI)
-                                        .name(name)
-                                        .build()
+                                    .protocol(ServiceProtocol.TCP) // TODO: default for the moment
+                                    .port(servicePort)
+                                    .targetPort(portI)
+                                    .name(name)
+                                    .build()
                         );
                     }
                 }

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -64,7 +64,7 @@
 
     <version.maven>3.3.1</version.maven>
     <version.fabric8>2.2.170</version.fabric8>
-    <version.docker-maven-plugin>0.16.7</version.docker-maven-plugin>
+    <version.docker-maven-plugin>0.16.9</version.docker-maven-plugin>
 
     <!-- Version of default images, updatable by CD -->
     <version.image.java.upstream.docker>1.2.1</version.image.java.upstream.docker>

